pkg/endpoint/policy.go

```diff
@@ -879,9 +884,68 @@ func (e *Endpoint) runIdentityToK8sPodSync() {
 	)
 }
 
+// This synchronizes the key-value store with a mapping of the endpoint's IP
+// with the numerical ID representing its security identity.
+func (e *Endpoint) runIPIdentitySync(endpointIP addressing.CiliumIP) {
+
+	if endpointIP == nil {
+		return
+	}
+
+	var addressFamily string
+	if endpointIP.IsIPv6() {
+		addressFamily = "ipv6"
+	} else {
+		addressFamily = "ipv4"
+	}
+
+	e.controllers.UpdateController(fmt.Sprintf("sync-%s-identity-mapping (%d)", addressFamily, e.ID),
+		controller.ControllerParams{
+			DoFunc: func() error {
+
+				ipKey := path.Join(ipcache.IPIdentitiesPath, ipcache.AddressSpace, endpointIP.String())
+				e.Mutex.RLock()
+				if e.SecurityIdentity == nil {
+					e.Mutex.RUnlock()
+					return nil
+				}
+				//identityValue := e.SecurityIdentity.ID.StringID()
+				ipIdPair := identityPkg.IPIdentityPair{
+					IP: endpointIP.IP(),
+					ID: e.SecurityIdentity.ID,
+				}
+
+				// Release lock as we do not want to have long-lasting key-value
+				// store operations resulting in lock being held for a long time.
+				e.Mutex.RUnlock()
+
+				marshaledIpIdPair, err := json.Marshal(ipIdPair)
+				if err != nil {
+					log.Errorf("err: %s", err)
+					return err
+				}
+
+				if err := kvstore.Update(ipKey, marshaledIpIdPair, true); err != nil {
+					return fmt.Errorf("unable to add endpoint IP '%s' to identity '%s': %s", ipKey, marshaledIpIdPair, err)
+				}
+				return nil
+			},
+			StopFunc: func() error {
+				ipKey := path.Join(ipcache.IPIdentitiesPath, ipcache.AddressSpace, endpointIP.String())
+				if err := kvstore.Delete(ipKey); err != nil {
+					return fmt.Errorf("unable to delete endpoint IP '%s': %s", ipKey, err)
+				}
+				return nil
+			},
+			RunInterval: 1 * time.Minute,
```

Comment:
This is too short, increase to 5 minutes

Reply:
@ianvernon Didn't we see a problem with a 5-minute interval? Like the first run of the controller happens sometimes too soon, before we know the labels for the endpoint, and in that case the endpoint doesn't get added to the identity for 5 minutes?

Reply:
@rlenglet no, that was a red-herring unfortunately.

Reply:
Increased to 5 minutes.